### PR TITLE
Add ADR white label architecture

### PR DESCRIPTION
**Pull request by wmontwe**
This ADR describes the white-label architecture to support two applications, namingly Thunderbird for Android and K-9 Mail, from one source repository.

### CODE REVIEW

**Comment by cketti** [comment]
We have quite a few modules named "common" in our project. To avoid confusion I think we should use Gradle's colon notation to refer to modules. So `:app-k9mail`, `:app-thunderbird`, and `:app:common`.

**Comment by cketti** [comment]
> At the moment it is hidden in the `app` module and I thought to give it more prominence by moving it to the project root. Maybe `app-common` would be beneficial.

I like that. I think `:app-common` would provide more context than simply naming it `:common`.

**Comment by wmontwe** [comment]
Update to `app-common`

File: docs/architecture/adr/0006-white-label-architecture.md [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+- Issue: [#7807](https://github.com/thunderbird/thunderbird-android/issues/7807)
+- Pull Request: [#7805](https://github.com/thunderbird/thunderbird-android/pull/7805)
+
+## Status
+
+- **Proposed**
+
+## Context
+
+Our project hosts two separate applications, K-9 Mail and Thunderbird for Android, which share a significant amount of functionality. Despite their common features, each app requires distinct branding elements such as app names, themes, and specific strings. To streamline our development process, a modular white-label approach is the desired approach. Each app is built as a module that relies on a shared codebase that should allows us to inject configuration details specific to each brand during the build or runtime. This will ensure that the core functionality remains consistent and maintainable.
```
Comment (cketti) [inline docs/architecture/adr/0006-white-label-architecture.md:12]: The last three sentences could probably be removed without losing any information. The information is repeated in the decision section (and that also feels like a better place for it).

```suggestion
Our project hosts two separate applications, K-9 Mail and Thunderbird for Android, which share a significant amount of functionality. Despite their common features, each app requires distinct branding elements such as app names, themes, and specific strings.
```

**Review by cketti — APPROVED** [review]

**Pull request merged**